### ISSUE TITLE
Users are resubscribed when using "cancel later" gist/code.

### ISSUE THREAD

**Issue by andrewlimaza** [strangerstudios/pmpro-mailchimp#84]
Users are being resubscribed, if there is code to give user's their level back to cancel on next payment date - https://www.paidmembershipspro.com/change-pmpro-membership-cancellation-to-set-expiration-date-for-next-payment-instead-of-cancelling-immediately/

This also is an issue with the Auto Renewal Add On.

**Comment by tictag** [comment]
Just to note an unintended consequence of this ... because the deleted user ends up being re-subscribed, MailChimp sends out our normal 'Welcome to the club!' email meaning that when we delete a user a few seconds later they get a welcome email!

Bit embarrassing for us.

**Comment by dparker1005** [comment]
This seems like an issue that should be addressed in the Gist itself. Maybe as a temporary workaround, we can do something like edit the Gist on affected sites so that it unhooks the MC process subscription changes function if the user is canceling and we're going to give them back their membership

**Comment by tictag** [comment]
For clarity, this issue is happening to my site and there is no gist (no custom code), just the Auto Renewal Add On. If your solution is to edit the originating code, then this add on will require a new release.

**Comment by dparker1005** [comment]
I have just set up the following test and was unable to re-create this issue:

*** Plugins - PMPro, PMProMC, PMProARC
*** Membership levels - only 1 level (Gold), recurring with monthly payments
*** PMPro Mailchimp settings - No double opt-in, Unsubscribe from old level audiences on level change, Gold membership subscribes to a Mailchimp list.

`
=== User Purchased Membership ===
[30-Jan-2020 18:28:01 UTC] Processing update for audience 482bf0a015: Array
(
    [0] => stdClass Object
        (
            [email_address] => dev-emai+Nicole.Wallace@flywheel.local
            [status] => subscribed
            [merge_fields] => Array
                (
                    [PMPLEVELID] => 1
                    [PMPALLIDS] => {1}
                    [PMPLEVEL] => Gold
                    [USERNAME] => Nicole.Wallace
                )

        )

)

=== User Canceled membership first time (and given one with expiration) ===
=== Both the unsubscription and re-subscription should be added to the update queue before the update is sent, so the unsubscription should be overwritten and never be sent ===
[30-Jan-2020 18:28:25 UTC] Processing update for audience 482bf0a015: Array
(
    [0] => stdClass Object
        (
            [email_address] => dev-emai+Nicole.Wallace@flywheel.local
            [status] => subscribed
            [merge_fields] => Array
                (
                    [PMPLEVELID] => 1
                    [PMPALLIDS] => {1}
                    [PMPLEVEL] => Gold
                    [USERNAME] => Nicole.Wallace
                )

        )

)

=== User cancels second time, correctly removed from audience ===
[30-Jan-2020 18:30:15 UTC] Processing update for audience 482bf0a015: Array
(
    [0] => stdClass Object
        (
            [email_address] => dev-emai+Nicole.Wallace@flywheel.local
            [status] => unsubscribed
            [merge_fields] => Array
                (
                    [PMPLEVELID] => 
                    [PMPALLIDS] => {}
                    [PMPLEVEL] => 
                    [USERNAME] => Nicole.Wallace
                )

        )

)`

Given how the MC batch update functionality is coded, this is what I would expect to happen and what did happen during my test. MC updates should only be pushed out on these hooks which, by default, do not run during the change membership level process: https://github.com/strangerstudios/pmpro-mailchimp/blob/95b65ca4e23199253f033f404d0ba6f472c45ada/pmpro-mailchimp.php#L1087-L1090

Happy to look further with this if you are still experiencing this issue on your site and can re-create a test that causes the issue. Feel free to reach out on Slack if you want to have more back and forth on this

**Comment by tictag** [comment]
**For mods only**
Please review https://www.paidmembershipspro.com/forums/topic/bug-mailchimp-add-on-member-is-left-subscribed-after-user-account-deleted/ for steps to recreate the issue, screenshots and debug log.

**Comment by dparker1005** [comment]
I've successfully re-created the issue where a deleted user with PMPro ARC sends a MC updated of `subscribed` instead of removing them from the mailing list. In my test where I set up the automatic welcome email, I did not receive the second welcome email when the user was deleted, likely because the subscribed user was being subscribed again (which makes sense).

Regardless, this seems like an issue that should be fixed in the PMPro ARC plugin since that seems to be where the conflicting behavior is originating, and may be causing conflicts in other plugins as well. Have created an Issue here that also links back to this thread: https://github.com/strangerstudios/pmpro-auto-renewal-checkbox/issues/14

Note that the Gist that is floating around for restoring cancelled recurring memberships also needs to be updated (also mentioned in the issue).
